exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -267,3 +293,63 @@ func timestampToSecondsWithMillisecondPrecision(ts pcommon.Timestamp) *float64 {
 func float64ToDimValue(f float64) string {
 	return strconv.FormatFloat(f, 'g', -1, 64)
 }
+
+// merge metric events to adhere to the multimetric format event.
+func mergeEventsToMultiMetricFormat(events []*splunk.Event) ([]*splunk.Event, error) {
+	merges := map[uint32][]int{}
+	hasher := fnv.New32a()
+	for i, e := range events {
+		cloned := copyEventWithoutValues(e)
+		data, err := jsoniter.Marshal(cloned)
+		if err != nil {
+			return nil, err
+		}
+		_, err = hasher.Write(data)
+		if err != nil {
+			return nil, err
+		}
+		hashed := hasher.Sum32()
+		hasher.Reset()
+		matches := merges[hashed]
+		if len(matches) == 0 {
+			merges[hashed] = []int{i}
+		} else {
+			merges[hashed] = append(merges[hashed], i)
+		}
+	}
+	var mergedEvents []*splunk.Event
+
+	for _, merge := range merges {
+		mergeDestination := events[merge[0]]
+		for j := 1; j < len(merge); j++ {
+			currentEvent := events[j]
+			// check no collision for this event values:
+			eventValues := map[string]interface{}{}
+			collisionFound := false
+			for field, value := range currentEvent.Fields {
+				if strings.HasPrefix(field, splunkMetricValue) {
+					if _, ok := mergeDestination.Fields[field]; ok {
+						// collision, same value. Stop iterating over the contents of the event.
```

Comment:
Does collision mean 2 identical metric data points with only one thing that can differ - value? That data is invalid in OTLP format. Do we really need all this code to support something considered an Undefined behavior? Maybe we can just ignore it and merge it with the latest value?

Reply:
Correct, this checks for that. Yes, if we don't check for collision, then we can simplify the logic quite a bit.